src/Aeon.Database/EnumerateColony.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Reactive.Linq;
+using Bonsai;
+using MySqlConnector;
+
+namespace Aeon.Database
+{
+    [Description("Enumerates all records in the colony table for each MySQL connection in the sequence.")]
+    public class EnumerateColony : Combinator<MySqlConnection, ColonyRecord>
+    {
+        public override IObservable<ColonyRecord> Process(IObservable<MySqlConnection> source)
+        {
+            return source.SelectMany(connection => ObservableDatabase.Query<ColonyRecord>(
+                "SELECT * FROM `#colony`;",
+                connection));
+        }
+    }
+
+    public class ColonyRecord
```

Comment:
just this one thing: did we decide how to handle this sort of type validation going forward? Or are we still gonna discuss it? Is there an issue / discussion referencing it? 

Otherwise, lgtm

Reply:
Do you mean the discussion on whether we generate the types from the DB vs validate explicit types against the DB? I don't think there is an issue for that one, I can create it.

In terms of how to handle a validation approach if we do go for it, that can also be changed and improved but wouldn't imply any breaking changes later.

Here I was going for a first usable prototype. We will have to resolve this more definitively once we start incorporating more tables from DJ into the acquisition, possibly at the time of the modular refactor, since the schemas could be aligned to match setup properties like camera locations, device serial numbers etc, that are already in the DB. It almost feels like this could all be synchronized somehow and we should discuss this, but yes I would say a separate issue.